pkg/controller/pattern1/deployment.go

```diff
@@ -317,13 +369,37 @@ func DashboardDeployment(apimanager *apimv1alpha1.APIManager, y *configvalues, n
 	labels := map[string]string{
 		"deployment": "wso2am-pattern-1-analytics-dashboard",
 	}
-	runasuser := int64(802)
+	//runasuser := int64(802)
 	//defaultMode := int32(0407)
 
 	dashVolumeMount, dashVolume := getAnalyticsDashVolumes(apimanager, num)
 
 	initContainers := getMysqlInitContainers(apimanager, &dashVolume, &dashVolumeMount)
 
+	dashbordSecurityContext := &corev1.SecurityContext{}
+	securityContextString := strings.Split(strings.TrimSpace(y.SecurityContext), ":")
+	securityContextType := securityContextString[0]
+
+	if securityContextType == "runAsUser" {
+		securityContextVal, _ := strconv.ParseInt(securityContextString[1], 10, 64)
+		dashbordSecurityContext.RunAsUser = &securityContextVal
+	} else if securityContextType == "runAsGroup" {
+		securityContextVal, _ := strconv.ParseInt(securityContextString[1], 10, 64)
+		dashbordSecurityContext.RunAsGroup = &securityContextVal
+	} else if securityContextType == "runAsNonRoot" {
+		securityContextVal, _ := strconv.ParseBool(securityContextString[1])
+		dashbordSecurityContext.RunAsNonRoot = &securityContextVal
+	} else if securityContextType == "privileged" {
+		securityContextVal, _ := strconv.ParseBool(securityContextString[1])
+		dashbordSecurityContext.Privileged = &securityContextVal
+	} else if securityContextType == "readOnlyRootFilesystem" {
+		securityContextVal, _ := strconv.ParseBool(securityContextString[1])
+		dashbordSecurityContext.ReadOnlyRootFilesystem = &securityContextVal
+	} else if securityContextType == "allowPrivilegeEscalation" {
+		securityContextVal, _ := strconv.ParseBool(securityContextString[1])
+		dashbordSecurityContext.AllowPrivilegeEscalation = &securityContextVal
+	}
+
```

Comment:
It looks like you have repeated the same code everywhere. So shall we add this code to a util function and then resolve the correct value from there.

Reply:
Done. Please review

Reply:
Fixed via [21d969b](https://github.com/wso2/k8s-wso2am-operator/pull/56/commits/21d969bb82db4e278c790520ff7eb2cfa3c94f24)